script/index.ts

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env npx ts-node
+import { promises as fs } from "fs";
+import { safeLoad } from "js-yaml";
+import { basename, extname, join } from "path";
+import { exec } from "./exec";
+
+interface WorkflowDesc {
+  folder: string;
+  id: string;
+}
+
+interface WorkflowsCheckResult {
+  compatibleWorkflows: WorkflowDesc[];
+  incompatibleWorkflows: WorkflowDesc[];
+}
+
+async function checkWorkflows(
+  folders: string[],
+  enabledActions: string[]
+): Promise<WorkflowsCheckResult> {
+  const result: WorkflowsCheckResult = {
+    compatibleWorkflows: [],
+    incompatibleWorkflows: [],
+  };
+
+  for (const folder of folders) {
+    const dir = await fs.readdir(folder, {
+      withFileTypes: true,
+    });
+
+    for (const e of dir) {
+      if (e.isFile()) {
+        const workflowFilePath = join(folder, e.name);
+        const enabled = await checkWorkflow(workflowFilePath, enabledActions);
+
+        const workflowDesc: WorkflowDesc = {
+          folder,
+          id: basename(e.name, extname(e.name)),
+        };
+
+        if (!enabled) {
+          result.incompatibleWorkflows.push(workflowDesc);
+        } else {
+          result.compatibleWorkflows.push(workflowDesc);
+        }
+      }
+    }
+  }
+
+  return result;
+}
+
+/**
+ * Check if a workflow only the given set of actions.
+ *
+ * @param workflowPath Path to workflow yaml file
+ * @param enabledActions List of enabled actions
+ */
+async function checkWorkflow(
+  workflowPath: string,
+  enabledActions: string[]
+): Promise<boolean> {
+  // Create set with lowercase action names for easier, case-insensitive lookup
+  const enabledActionsSet = new Set(enabledActions.map((x) => x.toLowerCase()));
+
+  try {
+    const workflowFileContent = await fs.readFile(workflowPath, "utf8");
+    const workflow = safeLoad(workflowFileContent);
+
+    for (const job of Object.keys(workflow.jobs || {}).map(
+      (k) => workflow.jobs[k]
+    )) {
+      for (const step of job.steps || []) {
+        if (!!step.uses) {
+          // Check if allowed action
+          const [actionName, _] = step.uses.split("@");
+          if (!enabledActionsSet.has(actionName.toLowerCase())) {
+            return false;
```

Comment:
What do you think about logging something here about the action that's causing this to not be included? Could be useful t to understand at a glance which other actions are used most.